config/config.go

```diff
@@ -204,5 +213,9 @@ func ReadConfig() Config {
 		MetricPartitioningRoutineInterval:      time.Duration(time.Duration(EnvOrDefaultInt(METRIC_PARTITIONING_ROUTINE_INTERVAL_SECONDS_ENVIRONMENT_KEY, DEFAULT_METRIC_PARTITIONING_ROUTINE_INTERVAL_SECONDS)) * time.Second),
 		MetricPartitioningRoutineDelayFirstRun: time.Duration(time.Duration(EnvOrDefaultInt(METRIC_PARTITIONING_ROUTINE_DELAY_FIRST_RUN_SECONDS_ENVIRONMENT_KEY, DEFAULT_METRIC_PARTITIONING_ROUTINE_DELAY_FIRST_RUN_SECONDS)) * time.Second),
 		MetricPartitioningPrefillPeriodDays:    EnvOrDefaultInt(METRIC_PARTITIONING_PREFILL_PERIOD_DAYS_ENVIRONMENT_KEY, DEFAULT_METRIC_PARTITIONING_PREFILL_PERIOD_DAYS),
+		RedisEndpointURL:                       os.Getenv(REDIS_ENDPOINT_URL_ENVIRONMENT_KEY),
+		RedisPassword:                          os.Getenv(REDIS_PASSWORD_ENVIRONMENT_KEY),
+		CacheTTL:                               time.Duration(EnvOrDefaultInt(CACHE_TTL_ENVIRONMENT_KEY, DEFAULT_CACHE_TTL_SECONDS)) * time.Second,
+		ChainID:                                os.Getenv(CHAIN_ID_ENVIRONMENT_KEY),
```

Comment:
"ChainID" has a very specific meaning in our context. seeing this made me think that it added a new restriction: the proxy service only works to field requests for one chain (ie. kava, chain id 2222). a simple read over of the config makes me think if i query for eth_chainID it will respond with this value.

however, this isn't the case. i'd propose renaming this to `CachePrefix` so it is clear that it can be used to invalidate the cache and that all keys are prefixed with it.

additionally, i'd add a validation that ensures it does not contain `:`, the cache key delimiter. if it does, some funky things could happen

Reply:
done